src/Components/CustomerDataPersistor/CustomerDataPersistor.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PayonePayment\Components\CustomerDataPersistor;
+
+use PayonePayment\RequestConstants;
+use Shopware\Core\Checkout\Customer\Aggregate\CustomerAddress\CustomerAddressEntity;
+use Shopware\Core\Checkout\Customer\CustomerEntity;
+use Shopware\Core\Checkout\Order\OrderEntity;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Validation\DataBag\RequestDataBag;
+use Shopware\Core\System\SystemConfig\SystemConfigService;
+
+class CustomerDataPersistor
+{
+    public function __construct(
+        private readonly EntityRepository $orderAddressRepository,
+        private readonly EntityRepository $customerRepository,
+        private readonly EntityRepository $customerAddressRepository,
+        private readonly SystemConfigService $configService
+    ) {
+    }
+
+    public function save(OrderEntity $order, RequestDataBag $dataBag, Context $context): void
+    {
+        $this->saveOrderData($order, $dataBag, $context);
+
+        if ($this->configService->getBool('PayonePayment.settings.saveUserEnteredDataToCustomer')) {
+            $customer = $this->getCustomer($order, $context);
+            if ($customer instanceof CustomerEntity) {
+                $this->saveCustomerData($customer, $dataBag, $context);
+                $customerAddress = $this->getCustomerAddress($customer, $context);
+                if ($customerAddress instanceof CustomerAddressEntity) {
+                    $this->saveCustomerAddressData($customerAddress, $dataBag, $context);
+                }
+            }
+        }
+    }
+
+    protected function saveCustomerData(CustomerEntity $customer, RequestDataBag $dataBag, Context $context): void
+    {
+        $birthday = $dataBag->get(RequestConstants::BIRTHDAY);
+        $birthday = \is_string($birthday) ? \DateTime::createFromFormat('Y-m-d', $birthday) ?: null : null;
```

Comment:
Can you put the `\DateTime::createFromFormat('Y-m-d', $birthday) ?: null` in brackets to make it easier to read?

Reply:
done.